### PR TITLE
Meddog

### PR DESCRIPTION
**Pull request by synt51**
predesigned style of the appointment page.
user appointments are only visible for the specific user

### CODE REVIEW

File: frontend/src/pages/MedDog.tsx
Comment (jamarob) [inline frontend/src/pages/MedDog.tsx:33]: Either uncomment or delete the function as it does nothing

File: frontend/src/pages/MedDog.tsx
Comment (jamarob) [inline frontend/src/pages/MedDog.tsx:19]: Does this function need to know that it is called on click action?

File: frontend/src/pages/MedDog.tsx
Comment (jamarob) [inline frontend/src/pages/MedDog.tsx:15]:
```suggestion
    const [open, setOpen] = useState(false);
    const [appointment, setAppointment] = useState<string>("")
    const [dateAndTime, setDateAndTime] = useState<Date | null>(new Date());
```
be consistent

File: frontend/src/pages/MedDog.tsx
Comment (jamarob) [inline frontend/src/pages/MedDog.tsx:41]: Headings are not for controlling font size. Using h4 breaks heading semantics.

File: frontend/src/pages/MedDog.tsx
Comment (jamarob) [inline frontend/src/pages/MedDog.tsx:70]:
```suggestion
                                onChange={setDateAndTime}
```

File: frontend/src/service/RequestService.ts
Comment (jamarob) [inline frontend/src/service/RequestService.ts:38]: Is this supposed to be called by an unauthorized user? If not make token required and remove the ternary

File: frontend/src/service/RequestService.ts
Comment (jamarob) [inline frontend/src/service/RequestService.ts:54]: delete comment

File: frontend/src/models/Appointment.ts
Comment (jamarob) [inline frontend/src/models/Appointment.ts:1]: You export only one thing from this file so this should be a default export

File: frontend/src/pages/MedDog.scss
Comment (jamarob) [inline frontend/src/pages/MedDog.scss:9]: Maybe research BEM for better class names http://getbem.com/

File: frontend/src/pages/MedDog.scss
Comment (jamarob) [inline frontend/src/pages/MedDog.scss:11]: Is this a flexbox or grid? 🧐 position: absolute without positional rules looks suspicious

File: frontend/src/components/AppointmentCard.tsx
Comment (jamarob) [inline frontend/src/components/AppointmentCard.tsx:28]: You have an interface for Appointments, maybe use it here?

File: frontend/src/components/AppointmentCard.tsx
Comment (jamarob) [inline frontend/src/components/AppointmentCard.tsx:29]: Can a user only have one appointment? Maybe we should use an id here?

File: frontend/src/components/AppointmentCard.tsx
Comment (jamarob) [inline frontend/src/components/AppointmentCard.tsx:20]:
```suggestion

```

File: frontend/src/components/AppointmentCard.tsx
Comment (jamarob) [inline frontend/src/components/AppointmentCard.tsx:23]: okay for now, but maybe add some error handling logic

File: frontend/src/components/AppointmentCard.tsx
Comment (jamarob) [inline frontend/src/components/AppointmentCard.tsx:9]:
```suggestion
    const [appointments, setAppointments] = useState<Appointment[]>([])
```

File: frontend/src/components/AppointmentCard.tsx
Comment (jamarob) [inline frontend/src/components/AppointmentCard.tsx:14]: Imho this belongs inside the RequestService

File: backend/src/main/java/de/neuefische/backend/service/MongoUserDetailsService.java
Comment (jamarob) [inline backend/src/main/java/de/neuefische/backend/service/MongoUserDetailsService.java:30]: Why pass the principal to the service you can use the above method just fine

File: backend/src/main/java/de/neuefische/backend/service/MongoUserDetailsService.java
Comment (jamarob) [inline backend/src/main/java/de/neuefische/backend/service/MongoUserDetailsService.java:24]: Imho it is confusing that the object es of type UserMongo and the service is called MongoUserDetailsService, be consistent with your names

File: backend/src/main/java/de/neuefische/backend/service/AppointmentsService.java
Comment (jamarob) [inline backend/src/main/java/de/neuefische/backend/service/AppointmentsService.java:31]: DTOs define your api and the service does not have to know about your api. Passing the entries from the dto as individual parameters leads to cleaner separation between your layers

File: backend/src/main/java/de/neuefische/backend/model/Appointments.java
Comment (jamarob) [inline backend/src/main/java/de/neuefische/backend/model/Appointments.java:27]: Why write a factory method when you have a builder?

File: backend/src/main/java/de/neuefische/backend/model/Appointments.java
Comment (jamarob) [inline backend/src/main/java/de/neuefische/backend/model/Appointments.java:18]: As I understand this is one apointment so maybe use singular form here?

File: backend/src/main/java/de/neuefische/backend/controller/AppointmentController.java
Comment (jamarob) [inline backend/src/main/java/de/neuefische/backend/controller/AppointmentController.java:31]: I don't understand the need for this method when `loadUserByUsername` throws `UsernameNotFoundException`

**Review by jamarob — COMMENTED** [review]
Nice! 🐕 

Left some suggestions for better separation of layers and to improve code quality. In some frontend files the formatting is a bit off, think about using prettier.

File: frontend/src/pages/MedDog.tsx
Comment (synt51) [reply frontend/src/pages/MedDog.tsx:33]: Line 32 is commented because the 'add Appointment' function is not yet fully written and there were errors when compiling.

File: frontend/src/pages/MedDog.tsx
Comment (synt51) [reply frontend/src/pages/MedDog.tsx:19]: You mean i should rename the function to just 'handleOpen'?

File: frontend/src/pages/MedDog.tsx
Comment (synt51) [reply frontend/src/pages/MedDog.tsx:15]: You're right, these are some Copy&Paste consts' from mui for the dialog.

File: frontend/src/service/RequestService.ts
Comment (synt51) [reply frontend/src/service/RequestService.ts:38]: The function should be callable by users who aren't logged in, but should immediately end in an error with an alert to log in before trying to safe any appointments. The alert part is missing because the function is not fully written atm.
Previously, i had some problems after making the token required in my 'createBagPlace' function, and because of that i copied my code of 'createBagPlace' to 'addAppointment

File: frontend/src/pages/MedDog.scss
Comment (synt51) [reply frontend/src/pages/MedDog.scss:11]: yeah, you're right. still need to fix all css files

File: frontend/src/components/AppointmentCard.tsx
Comment (synt51) [reply frontend/src/components/AppointmentCard.tsx:29]: Ups, you're right. specified any as type so that IntelliJ is quiet for the moment

File: frontend/src/components/AppointmentCard.tsx
Comment (synt51) [reply frontend/src/components/AppointmentCard.tsx:23]: yeah, the handling logic will be implemented later :)

File: frontend/src/components/AppointmentCard.tsx
Comment (synt51) [reply frontend/src/components/AppointmentCard.tsx:14]: Yeah, this will be fixed as soon as i clean up the code

File: backend/src/main/java/de/neuefische/backend/service/AppointmentsService.java
Comment (synt51) [reply backend/src/main/java/de/neuefische/backend/service/AppointmentsService.java:31]: I guess you're right.
I copied my 'createAppointment' from my 'createBagPlace' which i wrote similar with Florians' help.

File: backend/src/main/java/de/neuefische/backend/service/MongoUserDetailsService.java
Comment (synt51) [reply backend/src/main/java/de/neuefische/backend/service/MongoUserDetailsService.java:24]: Yeah, you're right, could be confusing. I will change that

File: backend/src/main/java/de/neuefische/backend/model/Appointments.java
Comment (synt51) [reply backend/src/main/java/de/neuefische/backend/model/Appointments.java:18]: You're right, will rename the class to the singular form

File: backend/src/main/java/de/neuefische/backend/model/Appointments.java
Comment (synt51) [reply backend/src/main/java/de/neuefische/backend/model/Appointments.java:27]: If I remember correctly, we learned it like that a few weeks ago. Since the i've been using these factory methods.
Would it just be shorter if I just the builder in the service or am i completely misunderstanding something?

Comment (jamarob) [reply backend/src/main/java/de/neuefische/backend/model/Appointments.java:27]: I would go with the builder as it is more expressive than a factory method - for the factory method you have to remember the order of arguments.

File: frontend/src/pages/MedDog.tsx
Comment (jamarob) [reply frontend/src/pages/MedDog.tsx:33]: okay, I am just not a huge fan of pushing stuff to main that is not working (yet).

File: backend/src/main/java/de/neuefische/backend/service/AppointmentsService.java
Comment (jamarob) [reply backend/src/main/java/de/neuefische/backend/service/AppointmentsService.java:31]: Don't worry that's not a big thing and it is perfectly okay to have different opinions on this. Just wanted to point it out.

**Review by jamarob — APPROVED** [review]
Hi Heinz! As I don't have write access to your repo an approval by me will not _unlock_ merging ability (maybe change your branch protection rule). I don't want to block this pr over little stuff any longer than needed. Maybe confer with @Flooooooooooorian about what is worth improving (maybe in another pr). Have a good one :v:

**Pull request merged**
